src/toil/provisioners/gceProvisioner.py

```diff
@@ -535,11 +535,16 @@ def addNodes(self, nodeType, numNodes, preemptable):
     def getProvisionedWorkers(self, nodeType, preemptable):
         entireCluster = self._getNodesInCluster(nodeType=nodeType)
         logger.debug('All nodes in cluster: %s', entireCluster)
-        workerInstances = [i for i in entireCluster if i.private_ips[0] != self.leaderIP]
+        workerInstances = []
+        for instance in entireCluster:
+            preemtible = False
+            scheduling = instance.extra.get('scheduling', None)
+            if scheduling and scheduling.get('preemptible', False):
+                preemtible = True
+            if instance.private_ips[0] != self.leaderIP and instance.state == 'running' and preemtible == preemptable:
```

Comment:
If instances have multiple private IPs wouldn't you want to check all of them instead of just the first one? So the first part would become `self.leaderIP not in instance.private_ips`?

Reply:
Toil only assigns one. This could be made more robust by checking for others.